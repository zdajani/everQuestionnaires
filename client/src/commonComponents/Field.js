import React from 'react';
import './styles/Field.css'

const renderField = ({ input, label, type, meta: { touched, error }, isTextArea }) => {
  const className = `form-group ${ touched && error ? 'has-danger' : '' }`;
  return(
    <div className={className}>
      {label &&  <label className="form-control-label">{label}</label>}
      
      {isTextArea ?
         <textArea {...input} placeholder={label} type={type} className="form-control"/> : 
         <input {...input} placeholder={label} type={type} className="form-control" />}
      
      <div className='text-help'>
        {touched && error && <span className="form-error">{error}</span>}
      </div>
    </div>
  );
}


export default renderField;