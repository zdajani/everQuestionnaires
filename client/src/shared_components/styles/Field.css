label.form-control-label {
  font-size: 1.3rem;
}

.form-error {
  color: #dc3545!important;
}

.form-control {
  height: 4rem;
}