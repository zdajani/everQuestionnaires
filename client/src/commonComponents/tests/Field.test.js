import React from 'react';
import { shallow } from 'enzyme';
import Field from '../Field';
import toJson from 'enzyme-to-json';

describe('Field component', () => {
  it('renders an input field', () => {
    const component = shallow(<Field 
      input="" 
      label="field testing"
      type="text"
      meta={ {touched: false, error: {} }} />);
    
    expect(toJson(component)).toMatchSnapshot();
  });
});