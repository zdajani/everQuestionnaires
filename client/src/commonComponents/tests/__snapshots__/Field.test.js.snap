// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component renders an input field 1`] = `
<div
  className="form-group "
>
  <label
    className="form-control-label"
  >
    field testing
  </label>
  <input
    className="form-control"
    placeholder="field testing"
    type="text"
  />
  <div
    className="text-help"
  />
</div>
`;
